### PR TITLE
fix: [RAC] Checkbox applies data attributes to both the input and the label

### PR DESCRIPTION
**Pull request by syd03098**
Closes 

* https://github.com/adobe/react-spectrum/issues/7070
* Do i need to add reviewers manually?

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

**Review by reidbarber — DISMISSED** [review]
Thanks for this!

File: packages/react-aria-components/src/Checkbox.tsx [hunk 1]
```diff
@@ -240,7 +239,10 @@ function Checkbox(props: CheckboxProps, ref: ForwardedRef<HTMLLabelElement>) {
       data-invalid={isInvalid || undefined}
       data-required={props.isRequired || undefined}>
       <VisuallyHidden elementType="span">
-        <input {...mergeProps(inputProps, focusProps)} ref={inputRef} />
+        <input
+          // data attribute should only go to the outermost dom
+          {...mergeProps(removeDataAttributes(inputProps), focusProps)}
```
Comment (yihuiliao) [inline packages/react-aria-components/src/Checkbox.tsx:244]: thanks for the pr!

i think this approach is fine but i think we want to match how we did it in other components like in this pr: https://github.com/adobe/react-spectrum/pull/5008. there's a [comment](https://github.com/adobe/react-spectrum/pull/5008#discussion_r1320225114) on that pr which mentions why

Comment (syd03098) [reply packages/react-aria-components/src/Checkbox.tsx:244]: ok, I will try that later :)

Comment (syd03098) [reply packages/react-aria-components/src/Checkbox.tsx]: Thanks for the review!

**Review by yihuiliao — APPROVED** [review]
lgtm! thanks for the test

**Review by devongovett — APPROVED** [review]

**Pull request merged**